plugins/Marketplace/resources/qml/OnboardBanner.qml

```diff
@@ -0,0 +1,138 @@
+
+// Onboarding banner.
+Rectangle
+{
+    property bool bannerVisible
+    property string bannerIcon
+    property string bannerBody
+    property var onRemoveBanner
+    property string readMoreUrl
+
+    visible: bannerVisible
+
+    Layout.preferredHeight: childrenRect.height + 2 * UM.Theme.getSize("default_margin").height
+    anchors
+    {
+        margins: UM.Theme.getSize("default_margin").width
+        left: parent.left
+        right: parent.right
+        top: parent.top
+    }
+
+    color: UM.Theme.getColor("action_panel_secondary")
+
+    // Icon
+    Rectangle
+    {
+        id: onboardingIcon
+        anchors
+        {
+            top: parent.top
+            left: parent.left
+            margins: UM.Theme.getSize("default_margin").width
+        }
+        width: UM.Theme.getSize("button_icon").width
+        height: UM.Theme.getSize("button_icon").height
+        color: "transparent"
+        UM.RecolorImage
+        {
+            anchors.fill: parent
+            color: UM.Theme.getColor("primary_text")
+            source: UM.Theme.getIcon(bannerIcon)
+        }
+    }
+
+    // Close button
+    UM.SimpleButton
+    {
+        id: onboardingClose
+        anchors
+        {
+            top: parent.top
+            right: parent.right
+            margins: UM.Theme.getSize("default_margin").width
+        }
+        width: UM.Theme.getSize("message_close").width
+        height: UM.Theme.getSize("message_close").height
+        color: UM.Theme.getColor("primary_text")
+        hoverColor: UM.Theme.getColor("primary_text_hover")
+        iconSource: UM.Theme.getIcon("Cancel")
+
+        onClicked: onRemoveBanner()
+    }
+
+    // Body
+    Label {
+        id: infoText
+        anchors
+        {
+            top: parent.top
+            left: onboardingIcon.right
+            right: onboardingClose.left
+            margins: UM.Theme.getSize("default_margin").width
+        }
+
+        font: UM.Theme.getFont("default")
+        text: bannerBody
+
+        renderType: Text.NativeRendering
+        color: "white"
+        wrapMode: Text.Wrap
+        elide: Text.ElideRight
+
+        onLineLaidOut:
+        {
+            if(line.isLast)
+            {
+                // Check if read more button still fits after the body text
+                if (line.implicitWidth + readMoreButton.width + UM.Theme.getSize("default_margin").width > width)
+                {
+                    // If it does place it after the body text
+                    readMoreButton.anchors.left = infoText.left;
+                    readMoreButton.anchors.bottom = infoText.bottom;
+                    readMoreButton.anchors.bottomMargin = -(fontMetrics.height + UM.Theme.getSize("thin_margin").height);
+                    readMoreButton.anchors.leftMargin = 0;
+                }
+                else
+                {
+                    // Otherwise place it under the text
+                    readMoreButton.anchors.left = infoText.left;
```

Comment:
In both cases of the if statement, the anchor is to the left. SO there is no need to set it in the onLineLaidOut. Same holds for the infotext.bottom

Reply:
https://github.com/Ultimaker/Cura/pull/10948/commits/b53c1ba3061b4fc3dbeb46e161f7f186fa399ceb